staging/src/k8s.io/apiextensions-apiserver/third_party/forked/celopenapi/model/types.go

```diff
@@ -89,13 +93,15 @@ func NewTypeParam(name string) *DeclType {
 		name:      name,
 		TypeParam: true,
 		exprType:  decls.NewTypeParamType(name),
+		celType:   cel.TypeParamType(name),
 	}
 }
 
-func newSimpleType(name string, exprType *exprpb.Type, zeroVal ref.Val) *DeclType {
+func newSimpleType(name string, exprType *exprpb.Type, celType *cel.Type, zeroVal ref.Val) *DeclType {
 	return &DeclType{
 		name:         name,
 		exprType:     exprType,
```

Comment:
Consider using `cel.TypeToExprType(celType)` to generate the `exprType` value.

Reply:
When using `el.TypeToExprType(celType)` to generate the exprType value, running into issue of empty list be treated as duration type instead of int.
The err received for test case`[].sum() == 0`: `compilation failed: ERROR: <input>:1:10: found no matching overload for '_==_' applied to '(duration, int)'`

Reply:
For that, you'll want to make sure that the `int`-based overload for sum appears as the first declaration provided to cel-go. You can be sure the overloads will be searched in order.

Reply:
We could switch to use `el.TypeToExprType(celType)` in the later release. Thanks for bringing this up!

Reply:
I don't think the call has anything to do with the error you're seeing though. To me that sounds like declaration order.

Reply:
The commit which cause the empty list failure for reference: https://github.com/kubernetes/kubernetes/commit/90844c629c6dbadabc263980c649635c42731dd5